### PR TITLE
fix(docs): publish real properties for object-shaped type aliases in SDK references

### LINKED ISSUE #4162: generated references: PostHogConfig publishes properties: []
## Bug description

the reference json that `generate-references` produces (and posthog.com renders for the sdk reference) has an empty shape for `PostHogConfig`: `"properties": []`, with the raw `Omit<BasePostHogConfig, 'loaded'> & {...}` text in `example`. the actual surface of that type is 132 keys, so the reference for the most-used type in the sdk renders nothing.

same thing hits the other object-shaped aliases (PostHogInterface, NetworkRecordOptions, OverrideConfig, FlagVariant, PropertyFilters, SiteApp/SiteAppGlobals/SiteAppLoader, SessionRecordingRemoteConfig/PersistedConfig, SurveyQuestion) — 52 of 107 types in the posthog-js references have empty properties overall, though most of those are string unions where the example string does the job. the dozen above are the ones losing real content. the parser is shared, so the siblings are affected too: posthog-node has 84 of 123 types with empty properties, react-native 73 of 115.

found it diffing the references json against the surface my extractor (github.com/ryanwaits/openpkg-ts) pulls from the checker.

## How to reproduce

1. `jq '.types[] | select(.name=="PostHogConfig")' packages/browser/references/posthog-js-references-latest.json` → `"properties": []`
2. `jq '[.types[] | select((.properties|length)==0)] | length' packages/browser/references/posthog-js-references-latest.json` → `52`
3. compare against the type itself: `packages/types/src/posthog-config.ts` has 130 top-level keys, plus the two the browser intersection adds in `packages/browser/src/types.ts`

## Related sub-libraries

- [ ] All of them
- [x] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [x] posthog-node
- [x] posthog-react-native
- [ ] @posthog/react
- [ ] @posthog/ai
- [ ] @posthog/convex
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin

## Additional context

root cause: `processTypeAliasMember` in `scripts/docs/types.js` runs `extractCallbackSignature` first, and its function heuristic is `signature.includes('=>') || signature.includes(': ')`. every object-shaped alias contains `': '`, so they all get classified as callback types — the raw signature lands in `example`, params gets emptied, and the early return skips property extraction entirely. interfaces are unaffected, which is why the method reference is in good shape.

happy to put up a PR: tighten the callback check to actual function-type structure and flatten Omit/intersections/object literals to their effective members via the checker

### PR DESCRIPTION
**Pull request by turnipdabeets**
## Problem

The generated SDK references published `"properties": []` for object-shaped type aliases — including `PostHogConfig`, the most-used type in the SDK — so posthog.com rendered nothing for them (fixes #4162, reported by @ryanwaits).

Root cause: `extractCallbackSignature` in `scripts/docs/types.js` treated any signature containing `': '` as a function type. Every object literal contains `': '`, so all object-shaped aliases took the callback path: raw signature into `example`, early return, property extraction skipped. Interfaces use a different path, which is why only aliases were affected. The parser is shared, so browser, node, and react-native references all had the gap (52/107, 95/124, 85/116 types with empty properties respectively).

This also unblocks PostHog/posthog.com#18577: the generated reference can now serve as the full `PostHogConfig` documentation source.

## Changes

- **New `scripts/docs/type-resolver.js`** — builds a TypeScript program over the same `.d.ts` entry point api-extractor analyzes and classifies each documented alias via the checker: `function`, `object` (flattened properties with JSDoc descriptions — this resolves `Omit<>`, intersections, and cross-package references, which token-based parsing can't), `signature` (tuples, index-signature-only objects), `union`, or `other`. Indexes entry-point exports plus every top-level alias in the program (api-extractor documents referenced-but-unexported aliases too); duplicate names across files resolve to `null` rather than guessing. `import("<abs path>").Name` qualifiers emitted by `typeToString` for non-lexically-visible symbols are stripped — they leak machine paths and would churn on every CI regeneration.
- **`scripts/docs/types.js`** — consults the resolver first; the `': '` heuristic is replaced with a top-level-arrow check on the fallback path; string-literal extraction no longer treats quoted strings inside `{ } & <` signatures (e.g. `Omit<T, 'key'>`) as union members.
- **`scripts/docs/utils.js`** — the fallback object parser skips index-signature parts instead of emitting `"[key"`-style property names.
- **`packages/{browser,node,react-native}/scripts/generate-docs.*`** — pass `dtsEntryPath` (identical to each package's api-extractor `mainEntryPointFilePath`).
- **Regenerated references**: `PostHogConfig` 0 → 132 properties with descriptions; empty-property types drop from 52→41 (browser), 95→49 (node), 85→43 (react-native). Remaining empties are string unions / interface unions rendered via `example`, unchanged from before.
- **Tests**: `scripts/docs/__tests__/` runs a fixture package through real api-extractor and the parser (24 node:test tests) covering object aliases, `Omit`+intersection, callbacks, string unions, tuples, index signatures, unions of interfaces, generic aliases, cross-file/unexported aliases, name collisions, import-qualifier stripping, and the resolver-less fallback. Wired into CI (`pnpm test:docs-scripts` in the unit job).

Reviewer-relevant verification: schema of the reference JSONs is unchanged (same keys); scripted before/after audit shows no type lost properties or example and no garbage property names; `packages/*/docs/*.api.json` and `.api.md` regenerate byte-identical; regeneration is deterministic (node's JSON byte-identical across runs). Union aliases still render as raw examples — expanding those via the checker is a possible follow-up, not a regression.

### Libraries affected

None — this is docs-generation tooling plus regenerated reference JSONs; no published package code changes, so no version bumps.

- [ ] All of them
- [ ] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [ ] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react-native-plugin
- [ ] @posthog/react
- [ ] @posthog/ai
- [ ] @posthog/convex
- [ ] @posthog/next
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/openfeature-node-provider
- [ ] @posthog/openfeature-web-provider
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types
- [ ] @posthog/browser-common

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [x] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [ ] Ran `pnpm changeset` to generate a changeset file

No changeset: nothing is released; docs tooling and generated reference files only.

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

- Built with Claude Code. Investigated #4162 against source, confirmed the reporter's root-cause analysis, and implemented the checker-based approach they proposed (thanks @ryanwaits).
- Considered `bundledPackages` in api-extractor config instead of a checker — rejected because it would rewrite the committed `.api.md`/`.api.json` public-API reports; the checker leaves those untouched.
- A pre-PR review loop (independent reviewer → skeptical triage → fix) caught two issues before this PR: `typeToString` leaking absolute local paths as `import("/Users/…")` qualifiers into the JSONs, and missing test coverage for duplicate-alias ambiguity. Both fixed; regeneration audits (schema keys, lost content, garbage names, determinism) ran after each cycle.

### CODE REVIEW

File: scripts/docs/type-resolver.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        return { kind: 'other' };
+    }
+
+    const properties = type.getProperties().filter((prop) => !(prop.flags & ts.SymbolFlags.Method));
+    if (type.getCallSignatures().length > 0 && properties.length === 0) {
+        return { kind: 'function' };
+    }
+    if (properties.length > 0) {
+        return {
+            kind: 'object',
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply scripts/docs/type-resolver.js:92]: Good catch — fixed in d3d525bfe: a callable type with members now returns `kind: 'signature'` and publishes the full raw signature (call signature included) instead of a property list; fixture + tests cover it. We scanned all documented aliases in the three packages with the checker and none currently hit this shape, so the committed reference JSONs are unchanged. Method-typed members on plain object aliases stay filtered deliberately — that matches how interfaces are rendered (property signatures only), and index signatures next to named properties follow the same interface precedent.

**Review by dustinbyrne — APPROVED** [review]
low context

i'm surprised `@microsoft/api-extractor` isn't doing all this by default / requires so much plumbing

**Comment by ryanwaits** [comment]
> low context
> 
> i'm surprised `@microsoft/api-extractor` isn't doing all this by default / requires so much plumbing

100% - ive hit this exact wall a bunch doing docs-quality work across ts packages, api-extractor stops at the declaration and the resolution plumbing is always hand rolled. i built an open spec that fixes a lot of this, was dogfooding it against posthog-js which revealed the gap
